extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java

```diff
@@ -278,6 +278,7 @@ private void onConnectionResumed() throws IOException, ApiException {
                 properties.put(Thing.PROPERTY_SERIAL_NUMBER, config.getMACAddress().replaceAll(":", "").toLowerCase());
                 properties.put(Thing.PROPERTY_FIRMWARE_VERSION, config.getSoftwareVersion());
                 updateProperties(properties);
+                updateThing(thing);
```

Comment:
How is this change related to the `AutomaticInboxProcessor registered as an RegistryListener`?

Reply:
If the properties would not be persisted, the automatic inbox listener would be unable to retrieve the representation property during deletion. Please note, that this was on purpose as long as the hue binding was updating the properties in each polling cycle. Now that it is guarded, that problem won't exist anymore and the properties should be persisted.

Reply:
Just came across this here. How does it relate to https://github.com/eclipse/smarthome/pull/1682 and esp. https://github.com/eclipse/smarthome/issues/588#issuecomment-226462285?

Reply:
@sjka I see your point but I wonder how to solve this issue without persisting the properties. As the thing is already gone when the removed callback is called I would not be able to get the property value. Hmm.

Reply:
@afuechsel I is always a bad smell to do some specific code changes in one binding for a concept that should be applicable in the whole framework (and thus be the same for all bindings). Note that after #1682 had been implemented, we meanwhile had https://github.com/eclipse/smarthome/pull/2629 in place. This allows to do Thing updates (structure AND configuration) and persists those updates in case it is a managed Thing. Imho, what is missing in that PR is that it also tries to persist properties in the very same way. If this was done, it would be in line with how configuration updates are handled and it would be exactly what you need here (i.e. this line can be removed as the framework already takes care of everything). Do you agree?

Reply:
I think, if properties would be persisted in some other way, this line could be removed. But did I understand correctly, that there is still some tiny piece missing (persisting the properties)? Would that mean to revert #1682?

Reply:
Yes. See #4688.

Reply:
Well, then this PR needs to be changed too.

Reply:
Yes. See #4688.

Reply:
Ah, didn't see it in the first place :)